libs/sdk-ui-kit/src/Dialog/StylingEditorDialog/StylingEditorDialog.tsx

```diff
@@ -0,0 +1,160 @@
+                            onChange={(e) => setNameField(e.target.value)}
+                        />
+                    </label>
+                    <label
+                        className={cx("gd-styling-editor-dialog-content-form-textarea", {
+                            "gd-styling-editor-dialog-content-form-invalid": invalidDefinition,
+                        })}
+                    >
+                        {intl.formatMessage({ id: "styling.editor.dialog.definition" })}
+                        <textarea
+                            className="gd-input-field s-textarea-field"
+                            wrap={"off"}
+                            value={definitionField}
+                            onChange={(e) => handleDefinitionFieldChange(e.target.value)}
+                        />
+                    </label>
+                    {invalidDefinition && (
+                        <Message
+                            className={cx(
+                                "gd-styling-editor-dialog-content-form-error",
+                                "s-gd-styling-editor-dialog-content-form-error",
+                            )}
+                            type="error"
+                        >
+                            <strong>Invalid definition.</strong> Please provide valid JSON object.
+                        </Message>
+                    )}
+                </form>
+                {providedExamples && (
+                    <div
+                        className={cx(
+                            "gd-styling-editor-dialog-content-examples",
+                            "s-gd-styling-editor-dialog-content-examples",
+                        )}
+                    >
+                        <BubbleHeaderSeparator
+                            title={intl.formatMessage({ id: "styling.editor.dialog.examples" })}
+                            message={tooltip}
+                        />
+                        <div className="gd-styling-editor-dialog-content-examples-list">
+                            {examples.map((example) => (
+                                <StylingExample
+                                    key={example.id}
+                                    name={example.title}
+                                    colors={exampleToColorPreview(example)}
+                                    onClick={() => {
+                                        setNameField(example.title);
+                                        setDefinitionField(JSON.stringify(example.theme, null, 4));
+                                    }}
+                                />
+                            ))}
+                        </div>
+                    </div>
+                )}
+            </div>
+            <StylingEditorDialogFooter
+                disableSubmit={isSubmitDisabled}
+                link={link}
+                onSubmit={() => (validateDefinition(definitionField) ? onSubmit : undefined)}
```

Comment:
dont you want to call onSubmit or change this to 
                onSubmit={validateDefinition(definitionField) ? onSubmit : undefined}
                
?